src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/SecurityController.php

```diff
@@ -0,0 +1,337 @@
+use PrestaShop\PrestaShop\Core\Domain\Security\Command\DeleteCustomerSessionCommand;
+use PrestaShop\PrestaShop\Core\Domain\Security\Command\DeleteEmployeeSessionCommand;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\CannotBulkDeleteCustomerSessionException;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\CannotBulkDeleteEmployeeSessionException;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\CannotDeleteCustomerSessionException;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\CannotDeleteEmployeeSessionException;
+use PrestaShop\PrestaShop\Core\Domain\Security\Exception\SessionNotFoundException;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShop\PrestaShop\Core\Form\FormHandlerInterface;
+use PrestaShop\PrestaShop\Core\Search\Filters\Security\Session\CustomerFilters;
+use PrestaShop\PrestaShop\Core\Search\Filters\Security\Session\EmployeeFilters;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Class SecurityController is responsible for displaying the
+ * "Configure > Advanced parameters > Security" page.
+ */
+class SecurityController extends FrameworkBundleAdminController
+{
+    /**
+     * Show sessions listing page.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @return Response
+     */
+    public function indexAction(): Response
+    {
+        $generalForm = $this->getGeneralFormHandler()->getForm();
+
+        return $this->render(
+            '@PrestaShop/Admin/Configure/AdvancedParameters/Security/index.html.twig',
+            [
+                'layoutHeaderToolbarBtn' => [],
+                'layoutTitle' => $this->trans('Security', 'Admin.Navigation.Menu'),
+                'generalForm' => $generalForm->createView(),
+            ]
+        );
+    }
+
+    /**
+     * Process the Security general configuration form.
+     *
+     * @AdminSecurity("is_granted(['update', 'create', 'delete'], request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function processGeneralFormAction(Request $request): RedirectResponse
+    {
+        return $this->processForm(
+            $request,
+            $this->getGeneralFormHandler(),
+            'General'
+        );
+    }
+
+    /**
+     * Process the Security configuration form.
+     *
+     * @param Request $request
+     * @param FormHandlerInterface $formHandler
+     * @param string $hookName
+     *
+     * @return RedirectResponse
+     */
+    protected function processForm(Request $request, FormHandlerInterface $formHandler, string $hookName): RedirectResponse
+    {
+        $this->dispatchHook(
```

Comment:
Why do you need to different hooks? And why the need for configurable `$hookName` when this method is only used once so `$hookName` is always to `General`

Reply:
Unless you just copied what was done in `AdministrationController` to anticipate there might be another form in this page?

Reply:
Because we plan to add other forms for the 1.7.9.0 related to password and security

Reply:
There is no 1.7.9 FYI